### PR TITLE
Bump Airbase to 126

### PR DESCRIPTION
**Pull request by ksobolew**
## Description

* Version 125 brings (among other things) updated dependency on Jackson, which fixes at least one CVE. Dependency on snakeyaml needs to be bumped to avoid conflicts with transitive dependencies.

* Version 126 brings fixes to how locale information is passed to Maven Surefire plugin. In relation to that, this commit adds a test to verify that the system locale is actually properly configured in the JVM.

> Is this change a fix, improvement, new feature, refactoring, or other?

Potential security fix, improvements to test stability.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Everything touching JSON

> How would you describe this change to a non-technical end user or system administrator?

Potential security fix, improvements to test stability.

## Related issues, pull requests, and links

airlift/airbase#314

airlift/airbase#315

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by ksobolew** [comment]
Created as draft without (yet) actually changing the version to see if the newly added test fails in CI. It does fail for me locally (my system locale is pl_PL).

File: pom.xml [hunk 1]
```diff
@@ -1679,7 +1679,7 @@
             <dependency>
                 <groupId>org.yaml</groupId>
                 <artifactId>snakeyaml</artifactId>
-                <version>1.28</version>
+                <version>1.30</version>
```
Comment (findepi) [inline pom.xml:1682]: Why?

**Comment by ksobolew** [comment]
```
Terminating due to java.lang.OutOfMemoryError: Java heap space
Error: Process completed with exit code 3.
```

☹️

**Comment by martint** [comment]
It's due to a broken version of the surefire plugin: https://github.com/trinodb/trino/pull/11900#issuecomment-1097736457

**Comment by findepi** [comment]
> It's due to a broken version of the surefire plugin: [#11900 (comment)](https://github.com/trinodb/trino/pull/11900#issuecomment-1097736457)

which was supposed to be fixed by https://github.com/airlift/airbase/pull/313.

now that's approved, we can perhaps merge it

File: pom.xml (hunk 1 above)
Comment (nineinchnick) [reply pom.xml:1682]: This was recently updated in #12091 but it looks like the latest jackson version requires this update: https://github.com/airlift/airbase/pull/314 and https://github.com/FasterXML/jackson-dataformats-text/commit/192f452efe62b9ec0d31cfd11ab1bab04b11eff4

**Comment by ksobolew** [comment]
`plugin/trino-phoenix` most likely has some initialization problem:

```
Caused by: io.trino.spi.TrinoException: line 1:135: Table 'phoenix.tpch.orders' does not exist
```

I seem to have seen this before.

https://github.com/trinodb/trino/runs/6770154443?check_suite_focus=true

**Comment by ksobolew** [comment]
> `plugin/trino-phoenix` most likely has some initialization problem:

Maybe #11344?

**Comment by ksobolew** [comment]
Are there plans to release new Airbase version @findepi?

**Comment by nineinchnick** [comment]
> Are there plans to release new Airbase version @findepi?

It was released yesterday. You can bump this to 126. There's also 127 but it'll require removing a few shaded org.apache imports, I have these changes in #12758 that's testing the latest version of Surefire.

**Review by findepi — APPROVED** [review]

**Comment by ksobolew** [comment]
CI hit: #12726

**Comment by ksobolew** [comment]
No other failures reported

**Review by nineinchnick — APPROVED** [review]

**Pull request merged**